app/src/main/java/de/danoeh/antennapod/fragment/DiscoveryFragment.java

```diff
@@ -116,28 +190,40 @@ public void onDestroy() {
         adapter = null;
     }
 
-    private void loadToplist() {
+    private void loadToplist(String country) {
         if (disposable != null) {
             disposable.dispose();
         }
+
         gridView.setVisibility(View.GONE);
         txtvError.setVisibility(View.GONE);
         butRetry.setVisibility(View.GONE);
         txtvEmpty.setVisibility(View.GONE);
         progressBar.setVisibility(View.VISIBLE);
 
-        ItunesTopListLoader loader = new ItunesTopListLoader(getContext());
-        disposable = loader.loadToplist(25).subscribe(podcasts -> {
+        if (country.equals(ItunesTopListLoader.DISCOVER_HIDE_FAKE_COUNTRY_CODE)) {
+            gridView.setVisibility(View.GONE);
+            txtvError.setVisibility(View.VISIBLE);
+            txtvError.setText(String.format(getResources().getString(R.string.discover_will_be_hidden),
+                    getResources().getString(R.string.discover_hide)));
+            butRetry.setVisibility(View.GONE);
+            txtvEmpty.setVisibility(View.GONE);
             progressBar.setVisibility(View.GONE);
-            topList = podcasts;
-            updateData(topList);
-        }, error -> {
-                Log.e(TAG, Log.getStackTraceString(error));
-                progressBar.setVisibility(View.GONE);
-                txtvError.setText(error.toString());
-                txtvError.setVisibility(View.VISIBLE);
-                butRetry.setOnClickListener(v -> loadToplist());
-                butRetry.setVisibility(View.VISIBLE);
-            });
+        } else {
+            ItunesTopListLoader loader = new ItunesTopListLoader(getContext());
+            disposable = loader.loadToplist(country, 25).subscribe(
+                    podcasts -> {
+                        progressBar.setVisibility(View.GONE);
+                        topList = podcasts;
+                        updateData(topList); },
```

Comment:
Oh, nice :) I did not know that this is the way to make Checkstyle happy about multiple lambda functions. I always used different indention for success and error.